File: solutions/python/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0: 
            return True 
        s_pointer = 0 
        t_pointer = 0 
        while t_pointer < len(t): 
            if t[t_pointer] == s[s_pointer]: 
                s_pointer += 1 
                if s_pointer == len(s): 
                    return True 
            t_pointer += 1 
        return False 
                                                                                                                                             

File: solutions/csharp/0392-is-subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        int j = 0;
        for (int i = 0; i < t.Length && j < s.Length; i++) {
            if (t[i] == s[j]) {
                j++;
            }
        }
        return j == s.Length;
    }
}
